fearless/Common/PurchaseProvider/MoonpayProvider.swift

```diff
@@ -26,22 +26,11 @@ final class MoonpayProvider: PurchaseProviderProtocol {
         return self
     }
 
-    func buildPurchaseActions(
-        for chain: Chain,
-        assetId _: WalletAssetId?,
-        address: String
-    ) -> [PurchaseAction] {
-        let optionUrl: URL?
-
-        guard chain == .polkadot else { return [] }
-        optionUrl = buildURLForToken("DOT", address: address)
-
-        if let url = optionUrl {
-            let action = PurchaseAction(title: "MoonPay", url: url, icon: R.image.iconMoonPay()!)
-            return [action]
-        } else {
-            return []
+    func buildPurchaseActions(asset: AssetModel, address: String) -> [PurchaseAction] {
```

Comment:
why array? is concrete provider can create multiple purchase actions?

Reply:
No. It's bad example of protocol realization, because each provider confirm it and return action, but also this protocol confirm PurchaseAggregator - some providers wrapper, which return all actions from all providers. I didn't have time to fix it.